xonsh/cli_utils.py

```diff
@@ -51,58 +94,289 @@ def get_doc(func: tp.Callable, parameter: str = None):
     str
         doc of the parameter/function
     """
+    if isinstance(func, str):
+        return func
     import inspect
 
     doc = inspect.getdoc(func) or ""
     if parameter:
-        return _get_param_doc(doc, parameter)
+        return inspect.cleandoc(_get_param_doc(doc, parameter)).strip()
     else:
-        return _get_func_doc(doc)
+        return _get_func_doc(doc).strip()
 
 
 _FUNC_NAME = "_func_"
 
 
+def _get_args_kwargs(annot: tp.Any) -> tp.Tuple[tp.Sequence[str], tp.Dict[str, tp.Any]]:
+    args, kwargs = [], {}
+    if isinstance(annot, tuple):
+        args, kwargs = annot
+    elif TYPING_ANNOTATED_AVAILABLE and "Annotated[" in str(annot):
+        if hasattr(annot, "__metadata__"):
+            args, kwargs = annot.__metadata__[0]
+        else:
+            from typing_extensions import get_args
+
+            _, (args, kwargs) = get_args(annot)
+
+    if isinstance(kwargs, tuple):
+        kwargs = dict(kwargs)
+
+    return args, kwargs
+
+
+def add_args(parser: ap.ArgumentParser, func: tp.Callable, allowed_params=None) -> None:
+    """From Arg -> parser.add_agument(*a, *kw)"""
```

Comment:
typo

Reply:
just explained that add_args is does what add_argument doing using `Arg` functions result. I think it doesn't do its job :)

Reply:
Yeah yeah the doc is good just a small typo, `agument` instead of `argument`

Reply:
oh just pushed changes for that to #4391